File: Leetcode/1099/1099.cpp
class Solution {
public:
    int twoSumLessThanK(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int n = (int)nums.size();
        int i = 0, j = n - 1;
        int maxSum = -1;
        int sum;
        while (i < j) {
            sum = nums[i] + nums[j];
            if (sum >= k) {
                --j;
            } else {
                maxSum = max(maxSum, sum);
                ++i;
            }
        }
        return maxSum;
    }
};

File: Leetcode/1099/1099.go
func twoSumLessThanK(nums []int, k int) int {
	sort.Ints(nums)
	i, j := 0, len(nums)-1
	sum := -1
	for i < j {
		temp := nums[i] + nums[j]
		if temp >= k {
			j--
		} else {
			if sum < temp {
				sum = temp
			}
			i++
		}
	}
	return sum
}